### PR TITLE
HTBHF-2474 Wire service into V2 controller and add integration tests.

### PR DESCRIPTION
**Pull request by mikeSmith08**
- Changed the existing controller tests to unit test the controller rather than be integration tests
 - Added integration tests against db and going straight to DWP.

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/dwp/controller/v2/DWPEligibilityControllerV2.java [hunk 1]
```diff
@@ -36,6 +39,9 @@ public IdentityAndEligibilityResponse getBenefits(@ApiParam("The identity and el
                                                               DWPEligibilityRequestV2 request) {
         log.debug("Received eligibility request");
 
-        return IdentityAndEligibilityResponse.builder().build();
+        IdentityAndEligibilityResponse response = service.checkIdentityAndEligibility(request);
+        log.debug("DWP identity status: {}, eligibility status: {}, qualifying benefits: {} ",
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/dwp/controller/v2/DWPEligibilityControllerV2.java]: I was unsure what to log here (if anything). The v1 controller simply outputs the eligibility response, but we have a lot more statuses on the response now so was unsure which to log. Don't want to log the whole object as it has children dobs on it. Thoughts?

**Review by gethyn1 — DISMISSED** [review]

File: api/src/main/java/uk/gov/dhsc/htbhf/dwp/controller/v2/DWPEligibilityControllerV2.java (hunk 1 above)
Comment (YetAnotherMatt) [reply api/src/main/java/uk/gov/dhsc/htbhf/dwp/controller/v2/DWPEligibilityControllerV2.java]: I think postcode & address matches would be useful as we then have all the statuses that determine whether someone has successfully identified themselves. Qualifying benefits may come in handy in the future, but right now they'll always be UC.

File: api/src/test/java/uk/gov/dhsc/htbhf/dwp/DWPIntegrationTestsV2.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    private ObjectMapper objectMapper;
+
+    @AfterEach
+    void clearDatabase() {
+        ucHouseholdRepository.deleteAll();
+        WireMock.reset();
+    }
+
+    @Test
+    void shouldReturnSuccessfulResponseWhenUserNotInDatabase() throws JsonProcessingException {
```
Comment (YetAnotherMatt) [inline api/src/test/java/uk/gov/dhsc/htbhf/dwp/DWPIntegrationTestsV2.java]: `shouldReturnStubResponseWhenUserNotInDatabase`?

**Review by YetAnotherMatt — DISMISSED** [review]

File: api/src/main/java/uk/gov/dhsc/htbhf/dwp/controller/v2/DWPEligibilityControllerV2.java (hunk 1 above)
Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/dwp/controller/v2/DWPEligibilityControllerV2.java]: Yep, sure thing, I'll add those then.

**Review by dwybourn — APPROVED** [review]

**Pull request merged**